crates/swift-bridge-ir/src/codegen/generate_c_header.rs

```diff
@@ -16,6 +16,11 @@ struct Bookkeeping {
     slice_types: HashSet<String>,
 }
 
+struct CFFiStructDeclarationBookkeeping {
```

Comment:
I think that most of the documentation that you put on `CFFiStruct` belongs here instead.

As in, we should document why we do this bookkeeping.